backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/JUnitBeansMap.java

```diff
@@ -121,11 +124,12 @@ public synchronized <T> ImmutableList<T> getBeansOfTypeOrNull(@NonNull final Cla
 	{
 		assertJUnitMode();
 
-		final List<Object> beanObjs = map.get(ClassReference.of(beanType));
-		if (beanObjs == null)
-		{
-			return null;
-		}
+		final List<Object> beanObjs = map.values()
+				.stream()
+				.filter(Objects::nonNull)
+				.flatMap(Collection::stream)
+				.filter(impl -> beanType.isAssignableFrom(impl.getClass()))
+				.collect(Collectors.toList());
```

Comment:
what's the point to collect to a list and then to stream again?

Reply:
also, what's the reason of this change?

Reply:
to be able to pick implementations of an interface